### ISSUE TITLE
[BUG]: Haunting Ground has artifacts using Vulkan renderer

### ISSUE THREAD

**Issue by EnekoT2001** [PCSX2/pcsx2#5340]
### Describe the Bug

When using half-pixel offset (necessary for this game to look good), there are vertical artifact lines on screen, most prominent in the secret room you can access after finishing the game. Using half.pizel offset texture or texture aggressive seems to make it less noticeable but the ideal config is normal vertex. Either way artifacts appear that don't appear on D3D11 or OGL.

### Reproduction Steps

Enable the half-pixel offset on any option except off (most noticeably the normal vertex one) and go to the secret room for example, also the title screen has a bit of artifacts too but not very noticeable.

### Expected Behavior

For the game to look fine/like on the OGL/D3D11 counterparts

### PCSX2 Revision

v1.7.2235

### Operating System

Windows 11

### CPU

Ryzen 3800X

### GPU

Radeon 6800XT

### GS Settings

Better to attach the inis in a zip
[HGCONFIG.zip](https://github.com/PCSX2/pcsx2/files/7874800/HGCONFIG.zip)

### Emulation Settings

I included them on the zip file mentioned above too
[HGCONFIG.zip](https://github.com/PCSX2/pcsx2/files/7874800/HGCONFIG.zip)

### GS Window Screenshots

![gs_20220115120126](https://user-images.githubusercontent.com/42989949/149619361-9b48a721-9c63-4759-8674-cdb23941f59d.png)
![gs_20220115120051](https://user-images.githubusercontent.com/42989949/149619363-8920125d-3277-4b00-82bf-540845526d92.png)
![gs_20220115120100](https://user-images.githubusercontent.com/42989949/149619364-6301dfb0-1421-472f-8584-1746ffd4c3e4.png)

### Logs & Dumps

[gs_dump.zip](https://github.com/PCSX2/pcsx2/files/7874805/gs_dump.zip)
[emuLog.txt](https://github.com/PCSX2/pcsx2/files/7874806/emuLog.txt)

**Comment by lightningterror** [comment]
Tried all options, unable to reproduce.
![image](https://user-images.githubusercontent.com/18107717/151961912-5469e97e-f5e5-4587-96f0-6785a9316fa6.png)

**Comment by EnekoT2001** [comment]
Sorr, couldn't reply faster, I had to fix my PC, I updated pcsx2 and I still have that bug, I uploaded a video on youtube to show it better. Did you try to use the exact settings I provided?

Or maybe it's an AMD GPU thing like many things. Also, another bug I saw was on some loading screens there seems to be a texture corruption? I think it's also bc vertex shader but didn't completely check, you can see it on the video too. Thing is, on my system the Vulkan rendered seems to be pretty broken still

https://youtu.be/hyPkMXH6QDc

**Comment by lightningterror** [comment]
Could probably be a bad setting, I see a lot of unnecessary stuff turned on/changed.

**Comment by EnekoT2001** [comment]
> Could probably be a bad setting, I see a lot of unnecessary stuff turned on/changed.

Idk tbh I’ve spent hours and hours trying every option in the past to get the best quality/graphical bugs fixed and performance, and on ogl/d3d there’s no issues, but I’ll try to tweak them a lot and see if I can figure out if another one is causing it then

**Comment by refractionpcsx2** [comment]
have you also made sure you're running the latest AMD drivers? We've seen a bunch of issues caused by old drivers (not AMD specifically)

**Comment by EnekoT2001** [comment]
> have you also made sure you're running the latest AMD drivers? We've seen a bunch of issues caused by old drivers (not AMD specifically)

Yeah, the latest version, also I just tried (on the latest pcsx2 version as of now) and I found out a thing. When half-screen fix is set to automatic, on Vulkan it gets enabled, and apparently, that causes the majority of the bug I've mentioned, but even if its set to force disabled, some dots/pixels appear at random where the lines would appear, it seems to me like the half-pixel offset option is inaccurate or a bit buggy and half-screen fix makes it much worse.

Also half screen fix seems to not do much on the graphics? never bothered to change it because automatic seems to work fine on other settings but well. Here's a video showing everything again https://youtu.be/LaWO0KIXIUg
